wgengine/router/router_linux.go: fixed " route del failed" error

c22wen opened the pull request:
`6.4M/24.1M router: route del failed: running “ip route del 100.100.100.100/32 dev tailscale0 table 52” failed: exit status 2` 

`ip addr del` is called before `ip route del` leading to error above. Switching order of `cidrDiff("addr")` and `cidrDiff("route")` to avoid error message.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/tailscale/tailscale/756)
<!-- Reviewable:end -->

danderson reviewed (commented):
Hmm, switching the order like this is liable to break route addition, isn't it? If removal needs to be route-then-addr, I would expect addition to require addr-then-route.

Do you still end up with all routes present and correct after restarting on this client?

c22wen commented:
> Hmm, switching the order like this is liable to break route addition, isn't it? If removal needs to be route-then-addr, I would expect addition to require addr-then-route.
> 
> Do you still end up with all routes present and correct after restarting on this client?

I think so. Didn't see any errors in the log when I did `tailscale up` and then `tailscale down`.

I tried it manually (`tailscaled`, then `ip route add`, `ip addr add`, `ip route del` and then `ip addr del`) and that didn't give me any issues either. I still see the addr I added with `ip addr list` even after I removed the route.
